core/src/storage2/lazy/lazy_cell.rs

```diff
@@ -416,4 +444,24 @@ mod tests {
             Ok(())
         })
     }
+
+    #[test]
+    fn set_works() {
+        let mut cell = <LazyCell<i32>>::new(Some(1));
+        cell.set(23);
+        assert_eq!(cell.get(), Some(&23));
+    }
+
+    #[test]
+    fn lazy_set_works() -> env::Result<()> {
+        run_test::<env::DefaultEnvTypes, _>(|_| {
+            let mut cell = <LazyCell<u8>>::lazy(Key::from([0x42; 32]));
+            let value = cell.get();
+            assert_eq!(value, None);
+
+            cell.set(13);
+            assert_eq!(cell.get(), Some(&13));
+            Ok(())
+        })
+    }
```

Comment:
Maybe some other test would be nice that test this functionality over the following:
1. Initialize a LazyCell<i32> with None
2. Push it to K
3. Pull another instance V from K, check that it is None
4. Set V using `set` to an actual value
5. Push V it to K
6. Load V2 from K
7. Check that V2 is the set value